create-event/src/App.js

```diff
@@ -0,0 +1,366 @@
+        .catch(err => console.log(err))
+    } else {
+      // FIXME: Pass all event attributes in PATCH
+      fetch(`${host}/event/${event.id}`, {
+        method: 'PATCH',
+        body: JSON.stringify({
+          data: {
+            attributes: event.attributes,
+            id: event.id,
+            type: 'event'
+          }
+        }),
+        headers: { Authorization: `Bearer ${apiKey}` }
+      })
+        .then(res => res.json())
+        .then(event => this.setState({ event: event.data }))
+        .catch(err => console.log(err))
+    }
+
+    tickets.map((ticket, index) => {
+      delete ticket.attributes.modified
+      // Convert from paid to free ticket if price is $0
+      const freeTicket = ticket.attributes.price === 0
+      if (freeTicket) {
+        ticket.attributes.type = 'free'
+      }
+
+      const newTicket = isNaN(ticket.id)
+
+      if (newTicket) {
+        fetch(`${host}/ticket_price`, {
+          method: 'POST',
+          body: JSON.stringify({ data: ticket }),
+          headers: { Authorization: `Bearer ${apiKey}` }
+        })
+          .then(res => res.json())
+          .then(response => (tickets[index] = response.data))
+          .catch(err => console.log(err))
+      } else {
+        fetch(`${host}/ticket_price/${ticket.id}`, {
+          method: 'PATCH',
+          body: JSON.stringify({ data: ticket }),
+          headers: { Authorization: `Bearer ${apiKey}` }
+        })
+      }
+      return true
+    })
+
+    deletedTickets.map(ticket => {
+      fetch(`${host}/ticket_price/${ticket.id}`, {
+        method: 'DELETE',
+        headers: { Authorization: `Bearer ${apiKey}` }
+      })
+      return true
+    })
+
+    this.snackbar(message)
+    this.setState({ tickets })
+    return true
+  }
+
+  activateEvent = () => {
+    const { event, tickets, apiKey } = this.state
+
+    const noTickets = tickets.length <= 0
+
+    if (!this.validation()) {
+      this.snackbar('Incomplete Form')
+      return false
+    }
+
+    if (noTickets) {
+      this.snackbar('Add Tickets')
+      return false
+    }
+
+    this.updateEvent('Updating...')
+
+    const activateReady = setInterval(() => {
+      const hasTickets = !isNaN(this.state.tickets[0].id)
+      const notLive = this.state.event.attributes.status !== 'active'
+
+      if (hasTickets && notLive) {
+        fetch(`${host}/event/${event.id}/activate`, {
+          method: 'POST',
+          headers: { Authorization: `Bearer ${apiKey}` }
+        })
+          .then(res => res.json())
+          .then(event =>
+            this.setState({ event: event.data }, () =>
+              this.snackbar('Event Live')
+            )
+          )
+        clearInterval(activateReady)
+      }
+    }, 1000)
+  }
+
+  validation = () => {
+    let validForm = true
+
+    this.state.tickets.map(ticket => {
+      const { name, price } = ticket.attributes
+
+      const badPrice = price < 3 && price > 0
```

Comment:
why check for price between 0 and 3? maybe switch to just less than 3, negative values can still be entered